src/topp/MapAlignerTreeGuided.cpp

```diff
@@ -0,0 +1,581 @@
+  static void computeTransformationsByOrigInFeature_(vector<FeatureMap>& feature_maps, FeatureMap& map_transformed,
+          vector<TransformationDescription>& transformations, const vector<Size>& trafo_order, const Param& model_params,
+          const String& model_type);
+  void computeTransformedFeatureMaps_(vector<FeatureMap>& feature_maps, const vector<TransformationDescription>& transformations);
+  void storeFeatureXMLs_(const vector<FeatureMap>& feature_maps, const StringList& out_files, FeatureXMLFile& fxml_file);
+  void storeTransformationDescriptions_(const vector<TransformationDescription>& transformations, StringList& trafos);
+
+
+  void registerOptionsAndFlags_() override
+  {
+    TOPPMapAlignerBase::registerOptionsAndFlags_("featureXML", REF_FLEXIBLE);
+    registerSubsection_("align_algorithm", "Algorithm parameters section");
+    registerSubsection_("model", "Options to control the modeling of retention time transformations from data");
+  }
+
+  Param getSubsectionDefaults_(const String&  section) const override
+  {
+    if (section == "align_algorithm")
+    {
+      MapAlignmentAlgorithmIdentification algo;
+      return algo.getParameters();
+    }
+    if (section == "model")
+    {
+      return TOPPMapAlignerBase::getModelDefaults("b_spline");
+    }
+    return Param(); // this shouldn't happen
+  }
+
+  ExitCodes main_(int, const char**) override
+  {
+    ExitCodes ret = checkParameters_();
+    if (ret != EXECUTION_OK) return ret;
+
+    //-------------------------------------------------------------
+    // parsing parameters
+    //-------------------------------------------------------------
+    StringList in_files = getStringList_("in");
+    StringList out_files = getStringList_("out");
+    StringList out_trafos = getStringList_("trafo_out");
+
+    Param model_params = getParam_().copy("model:", true);
+    String model_type = "b_spline";
+    model_params = model_params.copy(model_type+":", true);
+
+    //-------------------------------------------------------------
+    // reading input
+    //-------------------------------------------------------------
+    Size in_files_size = in_files.size();
+    FeatureXMLFile fxml_file;
+    // define here because needed to load and store
+    FeatureFileOptions param = fxml_file.getOptions();
+    // to save memory don't load convex hulls and subordinates
+    param.setLoadSubordinates(false);
+    param.setLoadConvexHull(false);
+    fxml_file.setOptions(param);
+
+    vector<FeatureMap> feature_maps(in_files_size);
+    loadInputMaps_(feature_maps, in_files, fxml_file);
+
+    //-------------------------------------------------------------
+    // calculations
+    //-------------------------------------------------------------
+
+    vector<vector<double>> maps_ranges(in_files_size);  // to save ranges for alignment (larger rt_range -> reference)
```

Comment:
can you move functions `buildTree_`, `treeGuidedAlignment_`, `computeTransformationsByOrigInFeature_` and `computeTransformedFeatureMaps_` to a class inside the library, so everyone can use it

Reply:
these should all be public (and hence tested in the class test)
`computeTransformationsByOrigInFeature_ ` could be renamed to `computeTrafosByOriginalRT_ `